pages/favorites.jsx

```diff
@@ -0,0 +1,80 @@
+import useLocalStorageState from "use-local-storage-state";
+import ActivityCard from "../components/ActivityCard";
+import styled from "styled-components";
+import useSWR from "swr";
+import { useEffect, useState } from "react";
+
+export default function Favorites() {
+  const [favoriteIds, setFavorites] = useLocalStorageState("favorites", {
+    defaultValue: [],
+  });
+
+  const {
+    data: activities,
+    error,
+    isLoading,
+  } = useSWR(
+    favoriteIds.length > 0 ? ["/api/activities/favourites", favoriteIds] : null,
+    async ([url, ids]) => {
+      const response = await fetch(url, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({ ids }),
+      });
+      if (!response.ok) {
+        throw new Error("Failed to fetch favorite activities");
+      }
+      return response.json();
+    }
+  );
+
```

Comment:
```suggestion
useEffect(() => {
  if (activities) {
    setLocalActivities(activities);
  }
}, [activities]);
```
Update the local state after the fetch